### PR TITLE
DOCS-20: Update data pipeline documentation

### PR DESCRIPTION
**Pull request by jonathan-fulton**
Adding updated documentation on our data pipeline. Replacing really out of date documentation from over a year ago.

### CODE REVIEW

File: docs/experiments/data-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+## Core Pipeline
+
+The core pipeline starts with the assignments, incrementally joins event data from your metrics/facts, aggregates the data, and computes winsorized aggregates.
+
+1. **Incremental daily data frame**: We join your experiment assignments to your fact sources. We use a "wide" table format where every metric aggregation is a separate column. As part of the incremental process, we first delete two days of data before re-inserting it. This is to ensure if there's time-delayed data it's updated appropriately.
+2. **Cumulative data frame**: We aggregate the daily data to generate a timeline view of all metric data for each subject.
+3. **Winsorized data frame**: We apply winsorization according to the cumulative data frame based on your metric settings.
+4. **Dimensionalized summaries**: We convert our subject-level winsorized data frame to an overall summary with dimensional values included.
+5. **Dimensionalized summaries reduced**: For high cardinality dimensions, we cap dimension values to 50 to minimize the data we transfer to our operational Postgres database.
```
Comment (aarsilv) [inline docs/experiments/data-pipeline.md:30]: How are the 50 chosen? The 50 most frequent?

File: static/img/experiments/data-pipeline/experiment_pipeline_visualization.png
Comment (aarsilv) [inline static/img/experiments/data-pipeline/experiment_pipeline_visualization.png:1]: Love the diagram!

**Review by aarsilv — APPROVED** [review]

File: docs/experiments/data-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 14
+---
+
+# Data pipeline
```
[the remark this answers is not in the document]
Comment (jonathan-fulton) [reply docs/experiments/data-pipeline.md:5]: Will clarify!

File: docs/experiments/data-pipeline.md (hunk 1 above)
Comment (jonathan-fulton) [reply docs/experiments/data-pipeline.md:30]: By assignment volume.  Will clarify.

**Pull request merged**